### DISCUSSION TITLE
How to reuse Pandoc extension "autolink_bare_uris" in quarto to revealjs conversion?

### DISCUSSION THREAD

**Discussion by nataschake** [quarto-dev/quarto-cli#6266] [Q&A] [answered]
### Description

Hi!
Pandoc extension "autolink_bare_uris" enabled automatically converts bare links, without writing them in <...>. 
Quarto does not do such a conversion: adding into _metadata.yml rows

```yaml
variant:
+autolink_bare_uris
format: 
  revealjs
```
and applying `quarto render presentation.qmd` I get
`The extension autolink_bare_uris is not supported for revealjs`.

The presentation.qmd looks like this:

```qmd
## Bare links

- In Pandoc: 
  - md_extension: +autolink_bare_uris
- In Quarto:
  - bare link is not clickable: https://www.wikidata.org/wiki/Q1
  - link in a code is not clickable: `https://www.wikidata.org/wiki/Q1`
  - [Clickable: reference written in GitHub Markdown style, gfm](https://www.wikidata.org/wiki/Q1)
  - Bare link surrounded with < > is clickable: <https://www.wikidata.org/wiki/Q1>
```
So the question is: how to autolink such bare links without rewriting the entire .qmd file with <> or []()?
Thanks.

**Comment by cderv** [comment, marked as answer]
Using `variant` will apply on the output format. You want the Pandoc's extension to be applied on the input format. 

https://quarto.org/docs/reference/formats/presentations/revealjs.html#rendering

So you need to do 
````markdown
---
title: "Statistical Learning"
format: revealjs
from: markdown+autolink_bare_uris
---

## Bare links

- In Quarto:
  - bare link is not clickable: https://www.wikidata.org/wiki/Q1
  - link in a code is not clickable: `https://www.wikidata.org/wiki/Q1`
  - [Clickable: reference written in GitHub Markdown style, gfm](https://www.wikidata.org/wiki/Q1)
  - Bare link surrounded with < > is clickable: <https://www.wikidata.org/wiki/Q1>

````

**Reply by TomBener** [reply]
I've tried this method and it works. But it seems not all EXTENSIONS work like this way in Quarto. For example:

```markdown
---
title: "Statistical Learning"
format: html
from: markdown+east_asian_line_breaks
---

测试一下
忽略
单个
中文
换行符号
```

Here [east_asian_line_breaks](https://pandoc.org/MANUAL.html#extensions) was applied to ignore newlines within a paragraph for East Asian wide characters.

For comparison, the following command worked well:

```sh
echo '忽略\n\n中文段落里\n单个换行符。ignore the newline\n within\n\na paragraph' | pandoc --from markdown+east_asian_line_breaks --to html
```

The output is as follows:

```
<p>忽略</p>
<p>中文段落里单个换行符。ignore the newline within</p>
<p>a paragraph</p>
```

Can you help to diagnose what's wrong with the issue? Thanks in advance.

**Reply by mcanouil** [reply]
@TomBener See:
- https://github.com/quarto-dev/quarto-cli/issues/8520
